### PR TITLE
Update: github actions, project name, cmake CMP0135

### PR DESCRIPTION
**Pull request by Yaraslaut**
* Updated github actions to include pedantic compilation and clang-tidy 
 * update third party cmake files: deleted unused libraries and update for [CMP0135](https://cmake.org/cmake/help/latest/policy/CMP0135.html)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,11 +13,11 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        cxx: [11,14,17, 20]
+        cxx: [17,20]
```
Comment (christianparpart) [inline .github/workflows/build.yml]:
```suggestion
        cxx: [17, 20]
```
Yeah I think it's totally fine to drop C++11 and C++14. People being interested in boxed-cpp most likely won't depend on a too old compiler anyways.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # C++ primitive type boxing
 
 This is a small header-only library for easing primitive type boxing in C++.
+Primary goal of the library is to make it easy to avoid code with easily swappable parameters [clang-tidy:bugprone-easily-swappable-parameters](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/easily-swappable-parameters.html?highlight=swappable).
+
+Overview on the topic: [C++ Weekly With Jason Turner](https://www.youtube.com/watch?v=Zq4yYPG7Erc)
```
Comment (christianparpart) [inline README.md:6]: 😮

File: README.md [hunk 3]
```diff
@@ -37,6 +40,50 @@ int main()
...
+
+``` c++
+#include <boxed-cpp/boxed.hpp>
+
+// Create unique structures
+namespace tags { struct Speed{}; struct Permittivity{}; struct Permeability{}; }
+
+using Speed = boxed::boxed<double,tags::Speed>;
+using Permittivity = boxed::boxed<double,tags::Permittivity>;
+using Permeability = boxed::boxed<double,tags::Permeability>;
```
Comment (christianparpart) [inline README.md]:
```suggestion
using Speed = boxed::boxed<double, tags::Speed>;
using Permittivity = boxed::boxed<double, tags::Permittivity>;
using Permeability = boxed::boxed<double, tags::Permeability>;
```

maybe it's just me, but i generally prefer a whitespace next to a comma.

Comment (christianparpart) [inline README.md]:
```suggestion
using Speed = boxed::boxed<double, tags::Speed>;
using Permittivity = boxed::boxed<double, tags::Permittivity>;
using Permeability = boxed::boxed<double, tags::Permeability>;
```

maybe it's just me, but i generally prefer a whitespace next to a comma.

File: README.md [hunk 4]
```diff
@@ -37,6 +40,50 @@ int main()
...
+        return wave_speed_inside(s, t);
+    }
+}
+
+
+int main()
+{
+    auto vacuum_permittivity = Permittivity(8.85418781762039e-12);
+    auto pi = 3.14159265358979323846;
+    auto vacuum_permeability = Permeability(4 * pi * 1e-7);
```
Comment (christianparpart) [inline README.md]: for the fun of it, all 3 of them could be made `constexpr`, right? ;-)

File: test-boxed-cpp.cpp [hunk 5]
```diff
@@ -53,8 +53,9 @@ TEST_CASE("boxed_cast with different inner types")
 {
     auto constexpr a = N{3};
     auto constexpr b = boxed_cast<Z>(a);
-
+#ifndef __GNUG__
```
Comment (christianparpart) [inline test-boxed-cpp.cpp:56]: why is this only working for GNU C++? 🤔 hmmmm

EDIT: oh wait. for all but GNU C++. Okay, all good. :D

Comment (Yaraslaut) [reply test-boxed-cpp.cpp:56]: > why is this only working for GNU C++? thinking hmmmm
> 
> EDIT: oh wait. for all but GNU C++. Okay, all good. :D

This is what g++ was telling me in github action
```
/home/runner/work/boxed-cpp/boxed-cpp/test-boxed-cpp.cpp:57:22: error: comparison of integer expressions of different signedness: ‘const unsigned int’ and ‘const int’ [-Werror=sign-compare]
   57 |     static_assert(*a == *b);
      |                   ~~~^~~~~
```

File: README.md (hunk 2 above)
Comment (Yaraslaut) [reply README.md:6]: > open_mouth

Do you want to change something in here ? :)

File: README.md (hunk 4 above)
Comment (Yaraslaut) [reply README.md]: > 

Done :)

File: README.md (hunk 2 above)
Comment (christianparpart) [reply README.md:6]: nop, i simply liked the links. as in: awesome! ;-)

**Review by christianparpart — APPROVED** [review]
many thanks, @Yaraslaut :)

**Pull request merged**